mne/beamformer/_lcmv.py

```diff
@@ -73,37 +74,105 @@ def _check_one_ch_type(info, picks, noise_cov):
         [_contains_ch_type(info_pick, tt) for tt in ('mag', 'grad', 'eeg')]
     if sum(ch_types) > 1 and noise_cov is None:
         raise ValueError('Source reconstruction with several sensor types '
-                         'requires a set of noise covariance matrices to be '
+                         'requires a noise covariance matrix to be '
                          'able to apply whitening.')
 
 
+def _pick_channels_spatial_filter(ch_names, spat_filt):
+    """Return data channel indices to be used with spatial filter.
+
+    Unlike ``pick_channels``, this respects the order of ch_names.
+    """
+    sel = list()
```

Comment:
ch_names = set(ch_names)

sel = []
for k, ch_name in spat_filt['ch_names']:
       if ch_name not in ch_names:
               raise ...
       else:
              sel.append(k)
return sel